pkg/controller/nodeconfig/sync_daemonsets.go

```diff
@@ -19,11 +21,14 @@ func (ncc *Controller) syncDaemonSet(
 	soc *scyllav1alpha1.ScyllaOperatorConfig,
 	daemonSets map[string]*appsv1.DaemonSet,
 ) error {
-	scyllaUtilsImage := soc.Spec.ScyllaUtilsImage
-	// FIXME: check that its not empty, emit event
-	// FIXME: add webhook validation for the format
+	if soc.Status.ScyllaDBUtilsImage == nil || len(*soc.Status.ScyllaDBUtilsImage) == 0 {
+		ncc.eventRecorder.Event(nc, corev1.EventTypeNormal, "MissingScyllaUtilsImage", "ScyllaOperatorConfig doesn't yet have scyllaUtilsImage available in the status.")
```

Comment:
shouldn't this result in Progressing condition instead? It's expected to be available eventually.

Reply:
I think for a different controller (NCC vs. SOC) this is a degraded state - it doesn't know whether the other controller can eventually make it, or is failing.
Effectively, it should be there from the operator start time, so emitting an event seems adequate.

Reply:
I added comment to wrong line, I meant to add Progressing condition instead returning an error which results in Degraded state.
I wouldn't call it degraded because you don't know if anything errored out, maybe it's just stale cache. Is stale cache a cause of object being degraded, I don't think so. In similar cases, our existing controllers are waiting with Progressing conditiong having set so this is inconsistent behavior.

Reply:
> maybe it's just stale cache. Is stale cache a cause of object being degraded, I don't think so

Right after you install the operator, this field should get set on the first sync loop of SOC. Or this one upgrade. If it's missing on other cases, something is broken - not a stale cache.

> In similar cases, our existing controllers are waiting with Progressing conditiong having set so this is inconsistent behavior.

That's not exactly the same - the case you provide w.r.t. our other resources is within a single controller and dependent objects while these are 2 independent controllers (also without any parent / child relationship).

A comparable case would be to say wanting to read a `kube-root-ca.crt` provisioned by Kubernetes controllers. If it's not there, you are Degraded, not Progressing as you are not making any progress to fixing that state (either directly or indirectly). (And you don't know whether the Kubernetes controllers are progressing toward the state, or Degraded.)

Reply:
I guess it's also a thin boundary so I can see a bit of point going both ways here.

Reply:
> That's not exactly the same - the case you provide w.r.t. our other resources is within a single controller and dependent objects while these are 2 independent controllers

What I meant for example, were cleanup Jobs waiting for HostID annotation reconciled by sidecar controller. These are two independent controllers.

> (also without any parent / child relationship).

I don't see how relationship matters. Nevertheless, ScyllaOperatorConfig is indirect parent of every resource reconciled by Operator, so they are related. Especially NodeConfigs, as since this PR SOC is part of its outputs.

> A comparable case would be to say wanting to read a kube-root-ca.crt provisioned by Kubernetes controllers. If it's not there, you are Degraded, not Progressing as you are not making any progress to fixing that state (either directly or indirectly). (And you don't know whether the Kubernetes controllers are Progressing toward the state, or Degraded.)

We don't do anything when tuning is running, nor we create cleanup jobs when HostID annotation is missing, yet we don't consider ScyllaCluster degraded but progressing.

There's artifical delay between when objects are created/updated when they need to be synchronized. That is expected and it's an error. In your case ConfigMap has a Namespace dependency so not having it immediately after Namespace creation is expected.